solr/solrj/src/java/org/apache/solr/client/solrj/impl/Http2SolrClient.java

```diff
@@ -203,6 +203,9 @@ protected Http2SolrClient(String serverBaseUrl, Builder builder) {
     } else {
       this.urlParamNames = Set.of();
     }
+    if (builder.executor != null) {
```

Comment:
*Move* code that is already initializing this in createHttpClient to here.  It includes the boolean to close it.